### PR TITLE
Bug 1907947: cloud-creds-secret creation with current context info only

### PR DESCRIPTION
**Pull request by nirarg**

### CODE REVIEW

**Comment by nirarg** [comment]
Hi @staebler ,
This PR is related to this comment: https://github.com/openshift/installer/pull/4350#discussion_r535833106

File: pkg/asset/installconfig/kubevirt/client.go [hunk 1]
```diff
@@ -59,9 +60,37 @@ var (
...
+	for k, v := range rawConfig.Contexts {
+		if k != rawConfig.CurrentContext {
+			delete(rawConfig.Contexts, k)
+		} else {
+			currentContextValue = v
+		}
+	}
+	if currentContextValue == nil {
+		return nil, fmt.Errorf("currentContext is not included in rawConfig.Contexts")
+	}
```
Comment (staebler) [inline pkg/asset/installconfig/kubevirt/client.go:81]:
```suggestion
	currentContextValue := rawConfig.Contexts[rawConfig.CurrentContext]
	if currentContextValue == nil {
		return nil, fmt.Errorf("currentContext is not included in rawConfig.Contexts")
	}
	rawConfig.Contexts = map[string]*clientcmdapi.Context{
		rawConfig.CurrentContext: currentContextValue,
	}
```

File: pkg/asset/installconfig/kubevirt/client.go [hunk 2]
```diff
@@ -59,9 +60,37 @@ var (
...
+	for k := range rawConfig.Clusters {
+		if k != currentContextValue.Cluster {
+			delete(rawConfig.Clusters, k)
+		}
+	}
```
Comment (staebler) [inline pkg/asset/installconfig/kubevirt/client.go:86]:
```suggestion
	if v, ok := rawConfig.Clusters[currentContextValue.Cluster]; ok {
		rawConfig.Clusters = map[string]*clientcmdapi.Cluster{
			currentContextValue.Cluster: v
		}
	}
```
Do a similar thing for AuthInfos, as well.

**Review by staebler — COMMENTED** [review]
Looks good. Just some nits.

/approve

File: pkg/asset/installconfig/kubevirt/client.go (hunk 2 above)
Comment (nirarg) [reply pkg/asset/installconfig/kubevirt/client.go]: Changed

File: pkg/asset/installconfig/kubevirt/client.go (hunk 1 above)
Comment (nirarg) [reply pkg/asset/installconfig/kubevirt/client.go]: Changed

**Pull request merged**